lms/djangoapps/discussion_api/api.py

```diff
@@ -624,20 +645,23 @@ def update_thread(request, thread_id, update_data):
         The updated thread; see discussion_api.views.ThreadViewSet for more
         detail.
     """
-    cc_thread, context = _get_thread_and_context(request, thread_id)
-    _check_editable_fields(cc_thread, update_data, context)
-    serializer = ThreadSerializer(cc_thread, data=update_data, partial=True, context=context)
-    actions_form = ThreadActionsForm(update_data)
-    if not (serializer.is_valid() and actions_form.is_valid()):
-        raise ValidationError(dict(serializer.errors.items() + actions_form.errors.items()))
-    # Only save thread object if some of the edited fields are in the thread data, not extra actions
-    if set(update_data) - set(actions_form.fields):
-        serializer.save()
-        # signal to update Teams when a user edits a thread
-        thread_edited.send(sender=None, user=request.user, post=cc_thread)
-    api_thread = serializer.data
-    _do_extra_actions(api_thread, cc_thread, update_data.keys(), actions_form, context)
-    return api_thread
+    try:
+        cc_thread, context = _get_thread_and_context(request, thread_id)
+        _check_editable_fields(cc_thread, update_data, context)
+        serializer = ThreadSerializer(cc_thread, data=update_data, partial=True, context=context)
+        actions_form = ThreadActionsForm(update_data)
+        if not (serializer.is_valid() and actions_form.is_valid()):
+            raise ValidationError(dict(serializer.errors.items() + actions_form.errors.items()))
+        # Only save thread object if some of the edited fields are in the thread data, not extra actions
+        if set(update_data) - set(actions_form.fields):
+            serializer.save()
+            # signal to update Teams when a user edits a thread
+            thread_edited.send(sender=None, user=request.user, post=cc_thread)
+        api_thread = serializer.data
+        _do_extra_actions(api_thread, cc_thread, update_data.keys(), actions_form, context)
+        return api_thread
+    except Http404:
```

Comment:
See my comment above about updating `_get_thread_and_context` instead of updating all its callers.